### ISSUE TITLE
Trailing comma in nested objects in interface

### ISSUE THREAD

**Issue by fourlastor** [palantir/tslint#3692]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7
- __Running TSLint via__: any

#### TypeScript code being linted

```ts
interface Foo {
  a: number
  b: {
    c: number // here linter warns a missing training comma
  }
}
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:recommended"
}
```

#### Actual behavior

Trailing comma is warned for `c: number`, interestingly enough if you extract an interface for `b` it doesn't.

#### Expected behavior

No lint warning.

**Comment by vovkasm** [comment]
Another sample:
```typescript
interface A {
  a: number
}
type B = A & {
  b: string // <-- here is incorrect "Missing trailing comma"
}
```

**Comment by jedmao** [comment]
The objects don't have to be nested. I've also seen it in this scenario:
```ts
function foo(
    bar: {
        baz,
        qux // linter warns missing trailing comma
    },
) {}
```

**Comment by adidahiya** [comment]
As per #4534, I'm closing issues related to formatting rules

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
